src/components/AvgMonthPaymentWidget/AvgMonthPaymentWidget.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Crm\PaymentsModule\Components;
+
+use Crm\ApplicationModule\Widget\BaseWidget;
+use Crm\ApplicationModule\Widget\WidgetManager;
+use Nette\Database\Connection;
+
+class AvgMonthPaymentWidget extends BaseWidget
+{
+    private $templateName = 'avg_month_payment_widget.latte';
+
+    private $database;
+
+    public function __construct(WidgetManager $widgetManager, Connection $database)
+    {
+        parent::__construct($widgetManager);
+        $this->database = $database;
+    }
+
+    public function identifier()
+    {
+        return 'avgmonthpaymentwidget';
+    }
+
+    public function render(array $params)
+    {
+        $segment = $params['segment'];
+        $data = $params['data'];
+        $usersIds = [];
+        $tableData = [];
+        $displayFields = false;
+
+        $segment->process(function ($row) use (&$usersIds, $data, &$tableData, &$displayFields) {
```

Comment:
As discussed outside this PR, you have the processed data available in the view which provides placeholder for this widget. Please reuse those data (https://github.com/remp2020/crm-segment-module/blob/master/src/presenters/StoredSegmentsPresenter.php#L137) instead of processing the segment again.

Reply:
Yes, I changed it after I read your comment but have not pushed it yet. All I need are those user IDs, so I will send them to the widgets ($tableData are not always available, sometimes the $data variable can be false but we still need to get the avg)